tests/aggregation/tests.py

```diff
@@ -988,6 +988,30 @@ def test_annotated_aggregate_over_annotated_aggregate(self):
         with six.assertRaisesRegex(self, FieldError, "Cannot compute Sum\('id__max'\): 'id__max' is an aggregate"):
             Book.objects.annotate(Max('id')).annotate(Sum('id__max'))
 
+        class MyMax(Max):
+            def as_sql(self, compiler, connection):
+                self.set_source_expressions(
+                    self.get_source_expressions()[0:1])
+                return super(MyMax, self).as_sql(compiler, connection)
+
+        with six.assertRaisesRegex(self, FieldError, "Cannot compute Max\(...'id__max'...\): 'id__max' is an aggregate"):
+            Book.objects.annotate(Max('id')).annotate(my_max=MyMax('id__max', 'price'))
+
+    def test_multi_arg_aggregate(self):
+        class MyMax(Max):
+            def as_sql(self, compiler, connection):
+                self.set_source_expressions(
+                    self.get_source_expressions()[0:1])
+                return super(MyMax, self).as_sql(compiler, connection)
+
+        with six.assertRaisesRegex(self, TypeError, 'Complex aggregates require an alias'):
```

Comment:
do you need `six.assertRaisesRegex` or would `self.assertRaisesMessage` work? (latter leaves less work to do when dropping Python 2).